### PR TITLE
Fixed replica reference for celery worker in Kubernetes.MD

### PR DESCRIPTION
**Pull request by zazathomas**
Fixed the replica reference for celery(beat & workers). The old reference doesn't exist based on the fields in the helm values file.
I've tested and the replicas are created as expected using the new reference. Also `redis.replicas` is not valid. The defectdojo helm chart doesn't use this field at all and only one replica would be created regardless. 
Willing to work with someone to fix further issues relating to K8S deployment as I plan to deploy defectdojo on K8s within the next few days

## :warning: Note on feature completeness :warning:

We are narrowing the scope of acceptable enhancements to DefectDojo in preparation for v3. Learn more here:
https://github.com/DefectDojo/django-DefectDojo/blob/master/readme-docs/CONTRIBUTING.md

**Description**

Describe the feature / bug fix implemented by this PR.
If this is a new parser, [the parser guide](https://documentation.defectdojo.com/contributing/how-to-write-a-parser/) may be worth (re)reading.

**Test results**

Ideally you extend the test suite in `tests/` and `dojo/unittests` to cover the changed in this PR.
Alternatively, describe what you have and haven't tested.

**Documentation**

Please update any documentation when needed in the [documentation folder](https://github.com/DefectDojo/django-DefectDojo/tree/dev/docs))

**Checklist**

This checklist is for your information.

- [ ] Make sure to rebase your PR against the very latest `dev`.
- [ ] Features/Changes should be submitted against the `dev`.
- [ ] Bugfixes should be submitted against the `bugfix` branch.
- [ ] Give a meaningful name to your PR, as it may end up being used in the release notes.
- [ ] Your code is flake8 compliant.
- [ ] Your code is python 3.11 compliant.
- [ ] If this is a new feature and not a bug fix, you've included the proper documentation in the docs at https://github.com/DefectDojo/django-DefectDojo/tree/dev/docs as part of this PR.
- [ ] Model changes must include the necessary migrations in the dojo/db_migrations folder.
- [ ] Add applicable tests to the unit tests.
- [ ] Add the proper label to categorize your PR.

**Extra information**

Please clear everything below when submitting your pull request, it's here purely for your information.

Moderators: Labels currently accepted for PRs:
- Import Scans (for new scanners/importers)
- enhancement
- performance
- feature
- bugfix
- maintenance (a.k.a chores)
- dependencies
- New Migration (when the PR introduces a DB migration)
- settings_changes (when the PR introduces changes or new settings in settings.dist.py)

# Contributors: Git Tips
## Rebase on dev branch
If the dev branch has changed since you started working on it, please rebase your work after the current dev.

On your working branch `mybranch`:
```
git rebase dev mybranch
```
In case of conflict:
```
 git mergetool
 git rebase --continue
 ```

When everything's fine on your local branch, force push to your `myOrigin` remote:
```
git push myOrigin --force-with-lease
```

To cancel everything:
```
git rebase --abort
```

## Squashing commits
```
git rebase -i origin/dev
```
- Replace `pick` by `fixup` on the commits you want squashed out
- Replace `pick` by `reword` on the first commit if you want to change the commit message
- Save the file and quit your editor

Force push to your `myOrigin` remote:
```
git push myOrigin --force-with-lease
```

### CODE REVIEW

File: readme-docs/KUBERNETES.md [hunk 1]
```diff
@@ -288,7 +288,8 @@ helm install \
   --set host="defectdojo.${TLS_CERT_DOMAIN}" \
   --set django.ingress.secretName="minikube-tls" \
   --set django.replicas=3 \
-  --set celery.replicas=3 \
+  --set celery.beat.replicas=3 \
```
Comment (kiblik) [inline readme-docs/KUBERNETES.md:291]:
```suggestion
```

File: readme-docs/KUBERNETES.md [hunk 2]
```diff
@@ -302,7 +303,8 @@ helm install \
   --namespace="${K8S_NAMESPACE}" \
   --set host="defectdojo.${TLS_CERT_DOMAIN}" \
   --set django.replicas=3 \
-  --set celery.replicas=3 \
+  --set celery.beat.replicas=3 \
```
Comment (kiblik) [inline readme-docs/KUBERNETES.md:306]:
```suggestion
```

**Review by kiblik — CHANGES_REQUESTED** [review]
Running multiple Workers to speed up the processing of tasks in the queue makes sense.
But running multiple Beats might have undesirable side-effects (like scheduling deduplication on different workers at the same time). The temporary unavailability of Beat does not degrade the quality of DD.

"A small delay in the scheduling of cleanup" or "the fact that alert would be sent a couple minutes later" is not as problematic as a situation when during deduplication one CeleryWorker would like to delete some finding that has been already removed by another Worker. 

FYI all celery beat tasks (Beat is responsible for scheduling, not for execution):
https://github.com/DefectDojo/django-DefectDojo/blob/d522bbe1b6edfec6612a29c005f85d3b8bbb2161/dojo/settings/settings.dist.py#L1117-L1159

**Comment by kiblik** [comment]
May I ask you to change the name of PR to a more descriptive form, please?

**Comment by zazathomas** [comment]
> Running multiple Workers to speed up the processing of tasks in the queue makes sense. But running multiple Beats might have undesirable side-effects (like scheduling deduplication on different workers at the same time). The temporary unavailability of Beat does not degrade the quality of DD.
> 
> "A small delay in the scheduling of cleanup" or "the fact that alert would be sent a couple minutes later" is not as problematic as a situation when during deduplication one CeleryWorker would like to delete some finding that has been already removed by another Worker.
> 
> FYI all celery beat tasks (Beat is responsible for scheduling, not for execution):
> 
> https://github.com/DefectDojo/django-DefectDojo/blob/d522bbe1b6edfec6612a29c005f85d3b8bbb2161/dojo/settings/settings.dist.py#L1117-L1159

Thanks @kiblik , that makes a lot of sense. I've committed the suggested changes and also fixed the title of the PR

**Comment by zazathomas** [comment]
@kiblik 
While I have you here, is there any documentation on how to use a separate postgres server using the helm deployment? I see the option to add a postgres server url but no details on the auth credentials required to connect to it. If you can point me in the right direction, I'd be more than happy to update the K8s guide with the details once I get it working. Thanks

**Comment by kiblik** [comment]
> @kiblik While I have you here, is there any documentation on how to use a separate postgres server using the helm deployment? I see the option to add a postgres server url but no details on the auth credentials required to connect to it. If you can point me in the right direction, I'd be more than happy to update the K8s guide with the details once I get it working. Thanks

If you would like to use Postgres deployed by Dojo helm chart, it might be a good idea to check https://github.com/bitnami/charts/tree/main/bitnami/postgresql/ and tune `postgres:` section of values. Unfortunately, I do not have the capacity to prepare it but I believe it would be beneficial for all Dojo users (who are using k8s deployments).

If you would like to use an external Postgres server, you are right, URL is one of the options. Dojo is loading EnvVars (related to DB) here: https://github.com/DefectDojo/django-DefectDojo/blob/d522bbe1b6edfec6612a29c005f85d3b8bbb2161/dojo/settings/settings.dist.py#L382-L404
So if you define `DD_DATABASE_URL`, all other params are ignored.
For format, check https://django-environ.readthedocs.io/en/latest/types.html#term-PostgreSQL.

`DD_...` envVars are injected:
- "directly" https://github.com/DefectDojo/django-DefectDojo/blob/d522bbe1b6edfec6612a29c005f85d3b8bbb2161/helm/defectdojo/templates/django-deployment.yaml#L181-L190
- from ConfigMap "semi-directly" https://github.com/DefectDojo/django-DefectDojo/blob/d522bbe1b6edfec6612a29c005f85d3b8bbb2161/helm/defectdojo/templates/django-deployment.yaml#L163-L165 e.g. https://github.com/DefectDojo/django-DefectDojo/blob/d522bbe1b6edfec6612a29c005f85d3b8bbb2161/helm/defectdojo/templates/configmap.yaml#L30-L34
- from ConfigMap "indirectly" from `.Values.extraConfigs` https://github.com/DefectDojo/django-DefectDojo/blob/d522bbe1b6edfec6612a29c005f85d3b8bbb2161/helm/defectdojo/templates/configmap.yaml#L48-L49 e.g. https://github.com/DefectDojo/django-DefectDojo/blob/d522bbe1b6edfec6612a29c005f85d3b8bbb2161/helm/defectdojo/values.yaml#L489
- from Secret "indirectly" from `.Values.extraSecrets` https://github.com/DefectDojo/django-DefectDojo/blob/d522bbe1b6edfec6612a29c005f85d3b8bbb2161/helm/defectdojo/templates/django-deployment.yaml#L166-L168 via https://github.com/DefectDojo/django-DefectDojo/blob/d522bbe1b6edfec6612a29c005f85d3b8bbb2161/helm/defectdojo/templates/extra-secret.yaml#L13-L16 e.g. https://github.com/DefectDojo/django-DefectDojo/blob/d522bbe1b6edfec6612a29c005f85d3b8bbb2161/helm/defectdojo/values.yaml#L487-L488
- from wherever you decide https://github.com/DefectDojo/django-DefectDojo/blob/d522bbe1b6edfec6612a29c005f85d3b8bbb2161/helm/defectdojo/templates/django-deployment.yaml#L205-L207  e.g. https://github.com/DefectDojo/django-DefectDojo/blob/d522bbe1b6edfec6612a29c005f85d3b8bbb2161/helm/defectdojo/values.yaml#L493-L498

**Review by kiblik — APPROVED** [review]

**Comment by zazathomas** [comment]
Thanks @kiblik 
I'll look into what you sent me and create a new PR with the detailed steps to help other K8s users struggling with similar issues.

**Comment by zazathomas** [comment]
@kiblik 
I've been able to get it working with an external postgres db. What's the preferred way of creating a PR for the guide? I could either add to the kuberntes.MD or create a dedicated .MD file. Let me know what you think

**Comment by kiblik** [comment]
> @kiblik I've been able to get it working with an external postgres db. What's the preferred way of creating a PR for the guide? I could either add to the kuberntes.MD or create a dedicated .MD file. Let me know what you think

I suppose @mtesauro is a better person to answer this question.

**Comment by mtesauro** [comment]
@zazathomas 
I'd say the "Useful stuff" section of https://github.com/DefectDojo/django-DefectDojo/blob/master/readme-docs/KUBERNETES.md#useful-stuff
since people will likely choose different ways to run PostgreSQL.

Thanks in advance for helping document those details! :+1:

**Comment by zazathomas** [comment]
@mtesauro 
No worries at all, glad to give back in anyway I can. I've created the PR [here](https://github.com/DefectDojo/django-DefectDojo/pull/10859). Is there anything else required on my part for these 2 PRs to be merged?
FYI @kiblik

**Comment by mtesauro** [comment]
@zazathomas One problem with your PR - we only take PRs against the Dev branch not master/main.

Mind pointing your PR against the Dev branch?  Then the reviews & approvals can start. :+1:

**Comment by zazathomas** [comment]
@mtesauro @kiblik 
It appears the PR has already been modified by one of the maintainers to point to the dev branch.

**Review by mtesauro — APPROVED** [review]
Approved

**Review by Maffooch — APPROVED** [review]

**Review by cneill — APPROVED** [review]

**Review by blakeaowens — APPROVED** [review]

**Pull request merged**